hpat/distributed_analysis.py

```diff
@@ -518,6 +518,14 @@ def _analyze_getitem(self, inst, lhs, rhs, array_dists):
                                                 new_dist.value))
             return
 
+        # array selection with permutation array index
+        if is_np_array(self.typemap, index_var.name):
+            arr_def = guard(get_definition, self.func_ir, index_var)
+            if (arr_def.op == 'call' and
+                    self._call_table[arr_def.func.name] == ['permutation', 'random', np]):
+                self._meet_array_dists(lhs, rhs.value.name, array_dists)
```

Comment:
`arr_def` could be None or any node (not necessarily Expr). This `if` needs more checks otherwise it fails.

Reply:
We need to check `rhs.value` to be multi-dimensional.

Reply:
call_table is gone now. need to use find_callname.